pkg/activator/handler/transport.go

```diff
@@ -0,0 +1,147 @@
+)
+
+func verifySanConnection(san string) func(tls.ConnectionState) error {
+	return func(cs tls.ConnectionState) error {
+		if cs.PeerCertificates == nil {
+
+			return errors.New("server certificate could not be verified during VerifyConnection. No PeerCertificates provided")
+		}
+		for _, name := range cs.PeerCertificates[0].DNSNames {
+			if name == san {
+				return nil
+			}
+		}
+		return fmt.Errorf("server with SAN %s does not have a matching name in certificate.  Names provided in certificate: %s", san, cs.PeerCertificates[0].DNSNames)
+	}
+}
+
+// tlsWrapper is a tls.Config wrapper with a TLS dialer for HTTP1
+type tlsWrapper struct {
+	tlsConf *tls.Config
+}
+
+// dialTLSContextHTTP1 handles HTTPS:HTTP1 dialer
+func (tw *tlsWrapper) dialTLSContextHTTP1(ctx context.Context, network, addr string) (net.Conn, error) {
+	return dialTLSContext(ctx, network, addr, tw.tlsConf)
+}
+
+// dialTLSContext handles HTTPS:HTTP1 and HTTP2 dialers
+// Depends on the activator handler's RevIDFrom
+func dialTLSContext(ctx context.Context, network, addr string, tlsConf *tls.Config) (net.Conn, error) {
+	config := activatorconfig.FromContext(ctx)
+	trust := config.Trust
+	if trust != netcfg.TrustDisabled {
+		tlsConf = tlsConf.Clone()
+		san := certificates.LegacyFakeDnsName
+		if trust != netcfg.TrustMinimal {
+			revID := RevIDFrom(ctx)
+			san = certificates.DataPlaneRoutingName(revID.Namespace)
+		}
+		tlsConf.VerifyConnection = verifySanConnection(san)
+	} else {
+		tlsConf = nil
+	}
+	return pkgnet.DialTLSWithBackOff(ctx, network, addr, tlsConf)
+}
+
+// newAutoTransport is a duplicate of the unexported knative/pkg's network newAutoTransport
+func newAutoTransport(v1, v2 http.RoundTripper) http.RoundTripper {
+	return pkgnet.RoundTripperFunc(func(r *http.Request) (*http.Response, error) {
+		t := v1
+		if r.ProtoMajor == 2 {
+			t = v2
+		}
+		return t.RoundTrip(r)
+	})
+}
+
+// newHTTPTransport is a duplicate of the unexported knative/pkg's network newHTTPTransport
+func newHTTPTransport(disableKeepAlives, disableCompression bool, maxIdle, maxIdlePerHost int) http.RoundTripper {
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.DialContext = pkgnet.DialWithBackOff
+	transport.DisableKeepAlives = disableKeepAlives
+	transport.MaxIdleConns = maxIdle
+	transport.MaxIdleConnsPerHost = maxIdlePerHost
+	transport.ForceAttemptHTTP2 = false
+	transport.DisableCompression = disableCompression
+	return transport
+}
+
+// Depends on the activator handler's RevIDFrom
+func newHTTPSTransport(disableKeepAlives, disableCompression bool, maxIdle, maxIdlePerHost int, tlsConf *tls.Config) http.RoundTripper {
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.DisableKeepAlives = disableKeepAlives
+	transport.MaxIdleConns = maxIdle
+	transport.MaxIdleConnsPerHost = maxIdlePerHost
+	transport.ForceAttemptHTTP2 = false
+	transport.DisableCompression = disableCompression
+
+	transport.TLSClientConfig = tlsConf
```

Comment:
Since you're setting DialTLSContext the godoc says TLSClientConfig isn't used so this is a noop